drivers/serial/serial_test.c

```diff
@@ -56,7 +146,14 @@ static int serial_vnd_init(const struct device *dev)
 }
 
 #define VND_SERIAL_INIT(n)                                                                         \
-	DEVICE_DT_INST_DEFINE(n, &serial_vnd_init, NULL, NULL, NULL, POST_KERNEL,                  \
+	COND_CODE_1(DT_INST_NODE_HAS_PROP(n, buffer_size),                                         \
```

Comment:
nit: This is a bit difficult to parse. I'd prefer if it was separated as two helper macros, decided by a single COND_CODE_1, where each makes their corresponding data struct definition and invocation of DEVICE_DT_INST_DEFINE.